include/zephyr/drivers/pinctrl/pinctrl_hpmicro_common.h

```diff
@@ -0,0 +1,47 @@
+/*
```

Comment:
any reason to not use `pinctrl_soc.h` directly? This is generally done when multiple SoCs share pinctrl drivers, like for example GD32 ARM/RISC-V SoCs.

Reply:
Yes, we plan to add another series of zephyr's chip pinctrl modules that will have some differences, this way we can ensure that we can reuse the driver.